test/integration/stats/data/service_controller_test.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Stats::Data::ServicesControllerTest < ActionDispatch::IntegrationTest
+  def setup
+    @application = FactoryBot.create :cinstance
+    host! @application.provider_account.admin_domain
+    user = @application.provider_account.admins.first!
+    @token = FactoryBot.create(:access_token, owner: user, scopes: %w[stats], permission: 'rw').value
+  end
+
+  attr_reader :application, :token
+
+  test 'top applications' do
+    params = { access_token: token, since: '2021-08-30 00:00:00', period: :day, metric_name: 'hits' }
+    get top_applications_stats_data_services_path(application.service, format: :json), params: params
+    assert_response :success
+  end
+
+  test 'missing params' do
+    params = { access_token: token, since: '2021-08-30 00:00:00', period: :day, metric_name: 'hits' }
+    get top_applications_stats_data_services_path(application.service, format: :json), params: params.except(:since)
+    assert_response :bad_request
+
+    get top_applications_stats_data_services_path(application.service, format: :json), params: params.except(:period)
+    assert_response :bad_request
```

Comment:
Can we have another test where `granularity` is present instead of `period`. Sorry for not being clear the first time.

From https://github.com/3scale/porta/blob/master/app/controllers/stats/api/base_controller.rb#L57
```ruby
permitted_params.require([:granularity, :since]) if permitted_params[:period].blank?
```